hw/ip/otbn/data/insns.yml

```diff
@@ -29,6 +29,353 @@ insn-groups:
     doc: |
       All Big Number (BN) instructions operate on the wide register file WREG.
 
+# Instruction encoding schemes
+#
+# These define the mapping between instruction operands and bits in the
+# encoding. A scheme names zero or more named fields. It can also inherit from
+# zero or more other schemes.
+#
+# The direct fields of a scheme are defined as a dictionary, mapping a field
+# name (which will be matched up with instruction operands) to a value. In
+# general, this value is itself a dictionary with the following keys:
+#
+#  bits: A list of ranges of bits. A range is written <msb>-<lsb>, where both
+#        are integers (and msb >= lsb). Multiple ranges can be separated by
+#        commas. A degenerate range (with msb == lsb) can be written as a bare
+#        integer. Required.
+#
+#  value: Optional. If specified, this should be a binary string for a fixed
+#         value for this field, prefixed with a "b" (to avoid the YAML parser
+#         reading it as a decimal number). Underscores in the string are
```

Comment:
Good point. I'll change to "0b" instead of "b".

Reply:
Gah. No, it turns out that the Python YAML parser parses 0b11 as 3. Since we need to keep leading zeros to catch width mismatches, can we just keep the `b11xx00` syntax for now?

Reply:
> can we just keep the b11xx00 syntax for now?

yes. 

(Even though I'd remove the width mismatch check and allow people just to specify numbers, as the field "value" would suggest to me.)

Reply:
Yeah, but it's more like a string whose alphabet happens to be `{0, 1}`, so `0010` is fundamentally a different thing from `10`. Or that's how I see it anyway.